internal/receiver/scriptedinputsreceiver/procpipe/config.go

```diff
@@ -0,0 +1,92 @@
+// Copyright Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procpipe
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+	"go.uber.org/zap"
+)
+
+// NewConfig creates a new stdin input config with default values
+func NewConfig() *Config {
+	return &Config{
+		InputConfig: helper.NewInputConfig(operatorType, operatorType),
+		BaseConfig: BaseConfig{
+			Multiline: helper.NewMultilineConfig(),
+			Encoding:  helper.NewEncodingConfig(),
+		},
+	}
+}
+
+// Config is the configuration of a stdin input operator.
+type Config struct {
+	helper.InputConfig `mapstructure:",squash"`
+	BaseConfig         `mapstructure:",squash"`
+}
+
+func (c *Config) Validate() error {
+	if c.ScriptName == "" {
+		return errors.New("'exec_file' must be specified")
+	}
+
+	_, ok := scripts[c.ScriptName]
+	if !ok {
+		return errors.New("Unsupported script " + c.ScriptName)
+	}
+
+	if c.MaxLogSize != 0 && c.MaxLogSize < minMaxLogSize {
+		return fmt.Errorf("invalid value for parameter 'max_log_size', must be equal to or greater than %d bytes", minMaxLogSize)
+	}
+	return nil
+}
+
+// BaseConfig is the detailed configuration of a tcp input operator.
+type BaseConfig struct {
```

Comment:
The component convention is to have a `scriptedinputsreceiver/config.go` file that contains the mapstructure tags that is used to populate the settings of any internal helper. There is generally also a `config_test.go` file that uses real yaml configs for validation and error testing. Please see https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/otlpreceiver and other receivers in contrib and this project for examples.

Reply:
If I move config to scriptedinputsreceiver directory I'd need to move also func (c *Config) Build(logger *zap.SugaredLogger) (operator.Operator, error) method from procpipe.go to parent directory and need access to Input struct which has private fields. I can either make these fields public or as one of my local colleagues suggested remove procpipe package all together and place everything in scriptedinputsreceiver. What do you suggest?

Reply:
Your requirement is based on the embedded, squashed config structs (simulating inheritance of multiple types) instead of composing the procpipe-specific fields and passing an operator instance.

Reply:
Current structure looks as follows
internal/receiver/scriptedinputsreceiver/procpipe/config.go
type BaseConfig struct {
	Multiline          helper.MultilineConfig `mapstructure:"multiline,omitempty"`
	ScriptName         string                 `mapstructure:"script_name,omitempty"`
	Encoding           helper.EncodingConfig  `mapstructure:",squash,omitempty"`
	Source             string                 `mapstructure:"source"`
	SourceType         string                 `mapstructure:"sourcetype"`
	CollectionInterval string                 `mapstructure:"collection_interval"`
	MaxLogSize         helper.ByteSize        `mapstructure:"max_log_size,omitempty"`
	AddAttributes      bool                   `mapstructure:"add_attributes,omitempty"`
}

type Config struct {
	helper.InputConfig `mapstructure:",squash"`
	BaseConfig         `mapstructure:",squash"`
}

internal/receiver/scriptedinputsreceiver/factory.go
func (f ReceiverType) CreateDefaultConfig() component.Config {
	return &ScriptConfig{
		BaseConfig: adapter.BaseConfig{
			Operators: []operator.Config{},
		},
		InputConfig: *procpipe.NewConfig(),
	}
}

Could you please provide more detailed description, maybe with some example how would you like me to reorganize the code as I'm still not clear what is expected.

Reply:
So I looked at this a bit and I think the issue may arise from the inner procpipe package not being necessary. By unifying the main `Config` around and embedded `helper.InputConfig` that builds `scriptedinputsreceiver` package level operator type the general conventions can be followed.  I found a few other issues in the process: https://github.com/rmfitzpatrick/splunk-otel-collector/blob/scripted_inputs_receiver/internal/receiver/scriptedinputsreceiver/config.go